src/Core/Indexes/Octree.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace AEGIS.Indexes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Represents a 3D Octree, which contains a collection of <see cref="IBasicGeometry" /> instances.
+    /// </summary>
+    public class Octree : QuadTree
+    {
+        /// <summary>
+        /// Represents a node of the Octree.
+        /// </summary>
+        protected class OctreeNode : QuadTreeNode
+        {
+            /// <summary>
+            /// The children nodes of this node. Each internal node must always have 8 children.
+            /// </summary>
+            private readonly List<OctreeNode> children;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="OctreeNode"/> class.
+            /// </summary>
+            /// <param name="envelope">The envelope of the node.</param>
+            public OctreeNode(Envelope envelope)
+                : base(envelope)
+            {
+                this.children = new List<OctreeNode>(8);
+            }
+
+            /// <summary>
+            /// Gets the children of the node.
+            /// </summary>
+            /// /// <value>The children of the node.</value>
+            public new List<OctreeNode> Children
```

Comment:
The `new` keyword hides the inherited method, and does not override it, therefore each time `Children` is used in the base class it will use the `Children` implementation in the `QuadTreeNode` class instead of this one. So this property will actually never be used (you can check it by debugging the octree tests and using breakpoints).